dataset_manager/__init__.py

```diff
@@ -1,50 +1,160 @@
+# -*- coding: utf-8 -*
+"""Dataset Manager
+
+This module helps to administrate the datasource
+for a DataScience projetc.
+
+"""
 import os
-import pandas as pd
+import logging
 import yaml
+from dataset_manager.data_source import DataSource
+logging.basicConfig(
+    level=logging.DEBUG,
+    format='%(asctime)s - '
+        '%(name)s - '
+        '%(funcName)s - '
+        '%(levelname)s - '
+        '%(message)s',
+    datefmt='%m-%d %H:%M')
+
+def _get_config_files(dataset_path):
+    all_files = os.listdir(dataset_path)
+    all_yaml = [yaml for yaml in all_files if yaml.endswith(".yaml")]
+    abs_yaml_files = [os.path.join(dataset_path, yaml_f) for yaml_f in all_yaml]
+    return abs_yaml_files
+
+def _parser_config_file(file):
+    result = {}
+    with open(file, "r") as conf_f:
+        ds_metadata = yaml.load(conf_f, Loader=yaml.FullLoader)
+        file_name = os.path.split(file)[-1]
+        identifier = file_name.split(".")[0]
+        result = {identifier : ds_metadata}
+    return result
+
+class DatasetManager:
+    """DatasetManager is the class to administrate the datasources
+    from a projec.
+
+    It is required a path with all datasource configurations as
+    yaml files.
+
+    Each file represents a datadource and must have the attributes:
+    `source`, `description`, `format` and
+    `local_source`(to save the downloaded data).
 
-class DatasetManager(object):
+    Args:
+        dataset_path: path to the datasets configurations.
+    """
     def __init__(self, dataset_path):
         self.__dataset_path = dataset_path
-    
-    def list_datasets(self):
```

Comment:
I renamed to `get_datasets`.
I removed all pandas dependencies, then the method `get_datasets` returns a dict instead of a DataFrame.
The `get_dataset` method helps the user to interact with the datasets configurations.

What do you think about a `show_datasets` method? It would print all datasets configurations.

Reply:
Great